### ISSUE TITLE
Answer for dif/2 with cyclic tree unsound due to incorrect display of variables

### ISSUE THREAD

**Issue by UWN** [trealla-prolog/trealla#210]
Somewhat forgotten part of [Item#14](http://www.complang.tuwien.ac.at/ulrich/trealla-prolog/Item#14)

**Comment by UWN** [comment]
```
?- -_=X,freeze(Y,X).
   X = -_A, freeze(Y,-_A).
?- - -_=X,freeze(Y,X).
   X = - -_A, freeze(Y,- -_A).
?- -X=X,freeze(Y,X).
   X = -X, freeze(_A,_B), unexpected, unsound.

**Comment by UWN** [comment]
Not really solved, but no longer an issue due to max_depth

**Comment by UWN** [comment]
Now:

```
?- dif(X,Y), -X=X.
   X = - - - - - - - - - ...,  unexpected.  % missing dif/2
```

**Comment by infradig** [comment]
Fixed in issue #313.
